src/Module/Api/Friendica/Events/Create.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Api\Friendica\Events;
+
+use Friendica\Core\Protocol;
+use Friendica\Core\Worker;
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Model\Event;
+use Friendica\Model\Conversation;
+use Friendica\Model\Item;
+use Friendica\Module\BaseApi;
+use Friendica\Network\HTTPException;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Worker\Delivery;
+
+/**
+ * API endpoint: /api/friendica/event_create
+ */
+class Create extends BaseApi
+{
+	protected function post(array $request = [])
+	{
+		BaseApi::checkAllowedScope(BaseApi::SCOPE_WRITE);
+		$uid = BaseApi::getCurrentUserID();
+
+		// params
+		$request = $this->getRequest([
+			'id'        => 0, //if provided, event will be amended
+			'name'      => '', //summary of the event
+			'desc'      => '', //description in BBCode
+			'startTime' => '', //starttime, required
+			'endTime'   => '', //endtime, required if nofinish false
+			'place'     => '', //location of the event
+			'publish'   => 0,  //publish message
+			'allow_cid' => '', //array of allowed person, if access restricted
```

Comment:
I prefer a unique name schema. So either `startTime` and `allowCid` or `start-time` and `allow-cid` or `start_time` and `allow_cid` and so on. I prefer the `-` and really dislike the `_`.

Reply:
This was normalized to `start_time` and `end_time` according to this: https://freunde.ma-nic.de/display/3e98eba8-2162-dc3d-20c5-cf3712415703